core/block/import/pb/gallery_test.go

```diff
@@ -0,0 +1,302 @@
+package pb
+
+import (
+	"testing"
+
+	"github.com/gogo/protobuf/types"
+	"github.com/stretchr/testify/assert"
+
+	"github.com/anyproto/anytype-heart/core/block/editor/state"
+	"github.com/anyproto/anytype-heart/core/block/editor/template"
+	"github.com/anyproto/anytype-heart/core/block/editor/widget"
+	"github.com/anyproto/anytype-heart/core/block/import/converter"
+	"github.com/anyproto/anytype-heart/pb"
+	"github.com/anyproto/anytype-heart/pkg/lib/bundle"
+	smartblock2 "github.com/anyproto/anytype-heart/pkg/lib/core/smartblock"
+	"github.com/anyproto/anytype-heart/pkg/lib/pb/model"
+	"github.com/anyproto/anytype-heart/util/pbtypes"
+)
+
+func TestGalleryImport_ProvideCollection(t *testing.T) {
+	t.Run("no widget in experience - root collection without objects", func(t *testing.T) {
+		// given
+		collectionProvider := GalleryImport{}
+		params := &pb.RpcObjectImportRequestPbParams{}
+
+		// when
+		collection, err := collectionProvider.ProvideCollection(nil, nil, nil, params, nil)
+
+		// then
+		assert.Nil(t, err)
+		assert.NotNil(t, collection)
+		rootCollectionState := state.NewDocFromSnapshot("", collection.Snapshot).(*state.State)
+		objectsInCollection := rootCollectionState.GetStoreSlice(template.CollectionStoreKey)
+		assert.Len(t, objectsInCollection, 0)
+	})
+	t.Run("CollectionTitle parameter is empty - collection with default name", func(t *testing.T) {
```

Comment:
**given** and **when** are identical in first and second usecases. Maybe we can unite the assertions?